src/Form/Form/RevisionJsonMenuNestedType.php

```diff
@@ -32,40 +32,39 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
     {
         /** @var FieldType $fieldType */
         $fieldType = $options['field_type'];
+        /** @var ContentType $contentType */
+        $contentType = $options['content_type'];
 
-        $builder->add('label', TextType::class, [
-            'constraints' => [new NotBlank()],
-        ]);
-
-        foreach ($fieldType->getChildren() as $child) {
-            /** @var FieldType $child */
-            if ($child->getDeleted()) {
-                continue;
-            }
-
-            $options = \array_merge([
-                'metadata' => $child,
-                'label' => false,
-            ], $child->getDisplayOptions());
-
-            $required = $child->getRestrictionOptions()['mandatory'] ?? false;
+        $labelChild = $fieldType->getChildren()->filter(fn (FieldType $c) => 'label' === $c->getName());
+        if (0 === $labelChild->count()) {
+            $labelFieldType = new FieldType();
+            $labelFieldType->setName('label');
+            $labelFieldType->setType(TextStringFieldType::class);
+            $labelFieldType->setOptions([
+                'displayOptions' => ['label' => 'Label'],
+                'restrictionOptions' => ['mandatory' => true],
```

Comment:
I think that the label doesn't have to be mandatory

Reply:
52de5b1